frontends/p4/fromv1.0/converters.cpp

```diff
@@ -176,6 +176,53 @@ const IR::Node* ExpressionConverter::postorder(IR::GlobalRef *ref) {
             new IR::Path(ref->srcInfo, IR::ID(ref->srcInfo, ref->toString())));
 }
 
+/// P4_16 is stricter on comparing booleans with ints
+/// Therefore we convert such expressions into simply the boolean test
+const IR::Node* ExpressionConverter::postorder(IR::Equ *equ) {
+  const IR::Expression *boolExpr = nullptr;
+  const IR::Expression *constExpr = nullptr;
+  if (equ->left->type->is<IR::Type_Boolean>() &&
+      (equ->right->type->is<IR::Type_InfInt>() && equ->right->is<IR::Constant>())) {
+    boolExpr = equ->left;
+    constExpr = equ->right;
+  } else if (equ->right->type->is<IR::Type_Boolean>() &&
+             (equ->left->type->is<IR::Type_InfInt>() && equ->left->is<IR::Constant>())) {
+    boolExpr = equ->right;
+    constExpr = equ->left;
+  }
+
+  if (boolExpr == nullptr)
+    return equ;
+
+  if (constExpr->to<IR::Constant>()->asInt() == 0)
```

Comment:
I followed the "C" convention: 0 is `false`, any other integer value is `true`.  Since we already have a boolean expression, testing it for equality with any other value than `0` or `1` is likely to be an error, so I should let it fail with the previous error by returning the original expression.